File: suitecrm-migration/backend/notification-service/src/main/java/com/suitecrm/notification/service/NotificationService.java
package com.suitecrm.notification.service;

import com.suitecrm.notification.dto.*;
import com.suitecrm.notification.entity.*;
import com.suitecrm.notification.repository.*;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import java.util.List;
import java.util.UUID;

@Service @RequiredArgsConstructor
public class NotificationService {
    private final AlertRepository alertRepository;
    private final FavoriteRepository favoriteRepository;

    public Page<AlertDto> getAlertsByUser(UUID userId, Pageable pageable) {
        return alertRepository.findByAssignedUserIdAndDeletedFalse(userId, pageable).map(this::toAlertDto);
    }
    public List<AlertDto> getUnreadAlerts(UUID userId) {
        return alertRepository.findByAssignedUserIdAndIsReadFalseAndDeletedFalse(userId).stream().map(this::toAlertDto).toList();
    }
    @Transactional
    public void markAlertRead(UUID id) {
        Alert alert = alertRepository.findByIdAndDeletedFalse(id)
                .orElseThrow(() -> new RuntimeException("Alert not found: " + id));
        alert.setIsRead(true);
        alertRepository.save(alert);
    }
    public List<FavoriteDto> getFavorites(UUID userId) {
        return favoriteRepository.findByAssignedUserIdAndDeletedFalse(userId).stream().map(this::toFavoriteDto).toList();
    }
    @Transactional
    public FavoriteDto addFavorite(UUID userId, String moduleName, UUID recordId) {
        Favorite fav = Favorite.builder().assignedUserId(userId).moduleName(moduleName).recordId(recordId).build();
        return toFavoriteDto(favoriteRepository.save(fav));
    }

    private AlertDto toAlertDto(Alert a) {
        return AlertDto.builder().id(a.getId()).name(a.getName()).alertType(a.getAlertType())
                .urlRedirect(a.getUrlRedirect()).targetModule(a.getTargetModule()).description(a.getDescription())
                .isRead(a.getIsRead()).assignedUserId(a.getAssignedUserId())
                .dateEntered(a.getDateEntered()).dateModified(a.getDateModified()).build();
    }
    private FavoriteDto toFavoriteDto(Favorite f) {
        return FavoriteDto.builder().id(f.getId()).moduleName(f.getModuleName())
                .recordId(f.getRecordId()).assignedUserId(f.getAssignedUserId()).dateEntered(f.getDateEntered()).build();
    }
}

File: suitecrm-migration/frontend/shell/src/app/services/notification.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Alert {
  id: string;
  name: string;
  alertType: string;
  urlRedirect: string;
  targetModule: string;
  description: string;
  isRead: boolean;
  dateEntered: string;
}

export interface Favorite {
  id: string;
  moduleName: string;
  recordId: string;
  assignedUserId: string;
  dateEntered: string;
}

@Injectable({ providedIn: 'root' })
export class NotificationService {
  private apiUrl = '/api/notifications';

  constructor(private http: HttpClient) {}

  getAlerts(userId: string, page = 0, size = 20): Observable<any> { return this.http.get(`${this.apiUrl}/alerts?userId=${userId}&page=${page}&size=${size}`); }
  getUnreadAlerts(userId: string): Observable<Alert[]> { return this.http.get<Alert[]>(`${this.apiUrl}/alerts/unread?userId=${userId}`); }
  markAlertRead(id: string): Observable<void> { return this.http.put<void>(`${this.apiUrl}/alerts/${id}/read`, {}); }
  getFavorites(userId: string): Observable<Favorite[]> { return this.http.get<Favorite[]>(`${this.apiUrl}/favorites?userId=${userId}`); }
  addFavorite(userId: string, moduleName: string, recordId: string): Observable<Favorite> {
    return this.http.post<Favorite>(`${this.apiUrl}/favorites?userId=${userId}&moduleName=${moduleName}&recordId=${recordId}`, {});
  }
}
